API/Controllers/SightController.cs

```diff
@@ -0,0 +1,65 @@
+﻿using BLL.Models;
+using BLL.Services.Contracts;
+using Microsoft.AspNetCore.Authorization;
+using Microsoft.AspNetCore.Mvc;
+
+namespace API.Controllers
+{
+    [ApiController]
+    [Authorize]
+    [Route("[controller]")]
+    public class SightController : ControllerBase
+    {
+        private readonly ISightService _service; 
+
+        public SightController(ISightService service)
+        {
+            _service = service;
+        }
+
+        [HttpGet]
+        public async Task<ActionResult<List<SightDTOModel>>> GetAllAsync()
+        {
+            return Ok(await _service.GetAllAsync());
+        }
+
+        [HttpGet("id")]
+        public async Task<ActionResult<SightDTOModel>> GetAsync(uint id)
+        {
+            var result = await _service.GetByIdAsync(id);
+            if (result == null)
+            {
+                return NotFound("Does not exist");
+            }
```

Comment:
it is illogical as for me. Firstly from name of your variable i can't undarstand what it is. 
Secondly check for null should be on bll. On bll you can throw exception and then cathc it in exception middleware